ui/components/component-library/text/text.types.ts

```diff
@@ -108,6 +108,10 @@ export interface TextProps extends BoxProps {
    * Changes the root html element tag of the Text component.
    */
   as?: ValidTag;
+  /**
+   * The id of the input associated with the label (in case Text is used as a Label)
+   */
+  htmlFor?: string;
```

Comment:
Going to work on a fixing the Box types to allow for polymorphic props so we can pass html attributes down to the html element without having to adding types to the original types. We can merge this and I'll work on it next.